apps/dashboard/app/models/project.rb

```diff
@@ -27,33 +27,69 @@ def find(project_path)
     def dataroot
       Rails.logger.debug("project path is: #{OodAppkit.dataroot.join('projects')}")
 
-      OodAppkit.dataroot.join('projects').tap do |p|
+      OodAppkit.dataroot.join('projects').tap do |path|
         p.mkpath unless p.exist?
       rescue StandardError => e
         Pathname.new('')
       end
     end
   end
 
+  validates :dir, presence: true
+  validates :dir, format: {
+
+    with: /[\w-]+\z/,
+    message: 'Name may only contain letters, digits, dashes, and underscores'
```

Comment:
```suggestion
    with: /^[\w-]+\z/,
    message: 'Directory may only contain letters, digits, dashes, and underscores'
```

Reply:
In order to get the tests to pass I used `\A` instead of `^` in order to clear out security errors in the integration tests.

Reply:
But, this also breaks the validation and it now accepts names like `kl;akje_kljdkj\\kljekj`

Reply:
Actually, it doesn't look like these validations are being applied at _all_, so I need to look into why that's happening.